commcare-cloud/commcare_cloud/commcare_cloud.py

```diff
@@ -89,14 +97,30 @@ def has_arg(unknown_args, short_form, long_form):
 
 
 class AnsibleContext(object):
-    def __init__(self):
+    def __init__(self, args):
         self._ansible_vault_password = None
+        self.env = self._build_env(args)
 
     def get_ansible_vault_password(self):
         if self._ansible_vault_password is None:
             self._ansible_vault_password = getpass.getpass("Vault Password: ")
         return self._ansible_vault_password
 
+    def get_environment(self):
+        return self.env
+
+    def _build_env(self, args):
+        """Look for args that have been flagged as environment variables
+        and add them to the env dict with appropriate naming
+        """
+        env = os.environ.copy()
+        env['ANSIBLE_CONFIG'] = os.path.expanduser('~/.commcare-cloud/ansible/ansible.cfg')
+        for arg, value in vars(args).items():
+            if arg.startswith(ENV_ARG_PREFIX) and value:
+                ansible_setting = arg[len(ENV_ARG_PREFIX):]
+                env['ANSIBLE_%s' % ansible_setting.upper()] = value
```

Comment:
I'm wondering if you can articulate why this is preferable to

```
if hasattr(args, 'env_stdout_callback'):
    env['ANSIBLE_STDOUT_CALLBACK'] = args.env_stdout_callback
```

That seems much clearer to me what's happening, but if you have a larger system you're working towards in your head then maybe this will make more sense to me.